akka-actor/src/main/scala/akka/actor/ActorCell.scala

```diff
@@ -421,6 +421,9 @@ private[akka] class ActorCell(
     with dungeon.DeathWatch
     with dungeon.FaultHandling {
 
+  private[this] var _props = _initialProps
+  def props: Props = _props
```

Comment:
The docs on `ActorContext.props` says `This method is thread-safe and can be called from other threads than the ordinary`, that is no longer true with this change, right? It also does not say anything about that props cannot be accessed from the context after the actor has terminated (which I think has changed in this PR).

Not sure if that is a show stopper, if it isn't it should probably be documented there?

Reply:
Ah, this is a good point that partly explains the above discussion :) I wonder if that's a useful feature, though? The problem is that to call it you have to close over the context which is discouraged and may fail as observed in the original ticket (where the context used to be yanked out under a stopped actor).

Reply:
Altogether, it's a wild mixture of methods that are marked thread-safe or not in ActorContext. For `props` in particular, it doesn't seem useful to explicitly state it that it is safe to be accessed from another thread, so we could remove that there.

Reply:
Yeah, I don't think of it as a feature that is very useful in particular, just that there was something we documented was threadsafe before and now we are changing that, in a patch release, not sure that is ok.

I don't know a reasonable scenario it would be used, perhaps someone could come up with their own backoff-re-starting-logic that wants to pick up props from ctx to restart the actor after it has stopped?